File: backend/Elearning.API/Models/Announcement.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Elearning.API.Models;

[Table("Announcement")]
[Index("IsPublished", Name = "IX_Announcement_IsPublished")]
public partial class Announcement
{
    [Key]
    public int AnnouncementId { get; set; }

    [StringLength(200)]
    public string Title { get; set; } = null!;

    public string Body { get; set; } = null!;

    public bool IsPublished { get; set; }

    public DateTime? PublishAt { get; set; }

    public DateTime? ExpiresAt { get; set; }

    public int CreatedByUserId { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public bool IsActive { get; set; }

    [ForeignKey("CreatedByUserId")]
    [InverseProperty("Announcements")]
    public virtual User CreatedByUser { get; set; } = null!;
}

File: frontend/elearning-web/src/types/announcement.ts
export type AnnouncementDto = {
  id: number;
  title: string;
  body: string;
  isPublished: boolean;
  publishAt: string | null;
  expiresAt: string | null;
  createdByUserId: number;
  userName?: string | null;
  createdAt: string;
  updatedAt: string | null;
  isActive: boolean;
};

export type AnnouncementCreateDto = {
  title: string;
  body: string;
  isPublished: boolean;
  publishAt?: string | null;
  expiresAt?: string | null;
};

export type AnnouncementEditDto = {
  id: number;
  title: string;
  body: string;
  isPublished: boolean;
  publishAt?: string | null;
  expiresAt?: string | null;
};
